### DISCUSSION TITLE
after update - fluid random splats enabled

### DISCUSSION THREAD

**Discussion by keshka** [rocksdanister/lively#2034] [General]
Am really enjoying the app. I had the spirit for some time. Last week Lively switched (I don't recall asking but may have clicked wrong button) and went to Fluid. I really liked it so it stayed and I have fun playing with it. Today, an announcement that an update had been installed, normally cool but now Fluid generates random splats in addition to splats from my mouse. I watched it for a while but it's a bit too busy while I work so I went to Customize Wallpaper and noticed the check box for Random Splats. It was not checked. Thinking this was what Fluid was doing I tried checking the box....no change, uncheck ...no change, reset to default....no change. Not sure what else to try. 

Suggestion, also add a variable for random splats that affects the frequency so we can adjust how often they occure.

Keshka Kotera
Summerville, Oregon

**Comment by rocksdanister** [comment]
Not sure how this happened, just remove this line to remove random splats completely:
https://github.com/rocksdanister/WebGL-Fluid-Simulation/blob/58d4b4c4111c1ee73c841b50da5bf44794b7d3e2/js/script.js#L65

**Reply by keshka** [reply]
No luck Dani;
I changed this file:
C:\Users\keshk\AppData\Local\Packages\12030rocksdanister.LivelyWallpaper_97hta09mmv6hy\LocalCache\Local\Lively Wallpaper\Library\wallpapers\0wj1biqk.f41\js

It's one of six script.js files that exist on my C: drive and it's what comes up when I right click on "Fluids" while Lively Wallpaper is open and I select "open file location".  I am a programmer, JS, sorta. I normally write in "C" but the JS structure is very similar so I'm pretty sure I did not damage anything.  I changed several things in the script.js file, one at a time. The changes have had no effect. It acts like I am modifying the wrong file but changing the file name kills Fluid so it must be correct. Here are the changed I made:

var timer = setInterval(randomSplat, 20000);	//Keshka mod changed from 3500
var _runRandom = false;			   	//Keshka mod changed to false 
var _isSleep = false;
function randomSplat()
{
//    if(_runRandom == true && _isSleep == false && _randomSplats)	Keshka mod, comment out
//        splatStack.push(parseInt(Math.random() * 20) + 5);		Keshka mod, comment out
}

Any other ideas sir?

**Reply by rocksdanister** [reply]
Oh maybe its the audio reactive component reacting to system audio?
You can comment it out:
https://github.com/rocksdanister/WebGL-Fluid-Simulation/blob/58d4b4c4111c1ee73c841b50da5bf44794b7d3e2/js/script.js#L77-L93

Apart from these two there is no other code that randomly generate splats.

**Reply by keshka** [reply]
Got it Dani!
my mod:

function multipleSplats (amount) {
    for (let i = 0; i < amount; i++) {
        const color = generateColor();
        color.r *= 10.0;
        color.g *= 10.0;
        color.b *= 10.0;
        const x = Math.random();
        const y = Math.random();
        const dx = 1000 * (Math.random() - 0.5);
        const dy = 1000 * (Math.random() - 0.5);
       // splat(x, y, dx, dy, color);		Keshka mod, comment out
    }
}

**Reply by keshka** [reply]
just saw your comment about the audio....I DO HAVE MUSIC ON! ...did not occur to me that might be the source
I will revisit that issue, if music caused it well that's cool! I will undo my mod. I like the random with music!

**Reply by keshka** [reply]
We used to have an old saying about code testing...."is that a feature or a flaw?" (grin)
